### PR TITLE
feat: add filter to table comment & make deleteRequest optional

### PR DESCRIPTION
**Pull request by ahmeterenodaci**
## How has this been tested?

- [ ] Prod
- [ ] Development

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] Added all translations, no hard coded text. 
- [ ] Updated the UI (if needed).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/ui/src/TableComponent/index.tsx [hunk 1]
```diff
@@ -38,11 +38,14 @@ export default function TableComponent({
   createOnNewPageTitle?: string;
   editOnNewPageUrl?: string;
   detailedFilter?: ColumnFilter[];
-  fetchRequest: (page: number) => Promise<{
+  fetchRequest: (
+    page: number,
+    filter?: { [key: string]: string | string[] },
```
Comment (a0m0rajab) [inline packages/ui/src/TableComponent/index.tsx:43]: Let's use the same type with filter here, having two different type is error prune

File: packages/ui/src/TableComponent/index.tsx [hunk 2]
```diff
@@ -55,9 +58,12 @@ export default function TableComponent({
   const [isLoading, setIsLoading] = useState(true);
   const isWindowExists = typeof window !== "undefined";
 
-  function getData(page: number) {
+  function getData(
+    page: number,
+    filter?: { [key: string]: string | string[] },
```
Comment (a0m0rajab) [inline packages/ui/src/TableComponent/index.tsx:63]: let's unify this type as well

**Review by a0m0rajab — CHANGES_REQUESTED** [review]

File: apps/web/src/app/[lang]/app/actions/table.ts [hunk 3]
```diff
@@ -159,6 +160,7 @@ export async function getTableData(
   type: TableDataTypes,
   page = 0,
   maxResultCount = 10,
+  filter?: Record<string, string | string[]>,
```
[the remark this answers is not in the document]
Comment (ahmeterenodaci) [reply apps/web/src/app/[lang]/app/actions/table.ts]: https://github.com/eren-ayasofyazilim/ayasofyazilim-ui/commit/35a459c4e49b2cc8ad9a75dd11509eca55bed714#diff-1a9afb2044dcfdbd0ca7ab01349da81aceeaec88cd0cea83e11c21cb5d600e40R331 I make an improvment for arrays. Now we use string or string array.

File: packages/ui/src/TableComponent/index.tsx (hunk 1 above)
Comment (ahmeterenodaci) [reply packages/ui/src/TableComponent/index.tsx]: done

File: packages/ui/src/TableComponent/index.tsx (hunk 2 above)
Comment (ahmeterenodaci) [reply packages/ui/src/TableComponent/index.tsx]: done

**Review by ertugrulcan-ays — APPROVED** [review]
lgtm

File: apps/web/src/app/[lang]/app/[type]/parties/table-data.tsx [hunk 4]
```diff
@@ -122,6 +122,26 @@ export const dataConfigOfParties = {
     translationKey: "Merchants" as const,
     subEntityName: "Merchants.SubOrganization" as const,
     subEntityType: "STORE" as const,
+    detailedFilters: [
+      {
+        name: "typeCodes",
+        displayName: "Type Codes",
+        type: "select-multiple" as const,
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/parties/table-data.tsx:129]: why did we need as const here?

**Review by a0m0rajab — APPROVED** [review]

**Pull request merged**